### ISSUE TITLE
Introduce in-game lobbies

### ISSUE THREAD

**Issue by nvs** [nvs/gem#116]
Useful to stay in a game with friends who don't want to remake the game for just one player. Or for helping new players who are being instructed, but lose fairly quickly.

Would have to ensure that restarting cannot be abused by people who simply want to get a one hit on the first couple rounds or so.

**Comment by nvs** [comment]
Probably want the following:
- A secondary timer for each player showing the time for that particular session (assuming they restarted).
- An indication on the board that a player has restarted.
- The ability (once board customization is possible) to show/hide session times of other players.
- The ability to enable/disable restarting as a setting.

**Comment by nvs** [comment]
This is blocked by a lot. Some of the topics are not covered in issues yet.

**Comment by nvs** [comment]
The concept of this topic roughly remains the same. But the goals and scope will change quite a bit. For starters, it will be more about creating in-game lobbies that players can join. These can be private or public, be created on demand, and others will be able to join them. The goal is to let people join and continue playing until the lobby they are in has started. At that point, they will start a fresh game.

As for implementation details, I'm thinking the following:
- A tavern/shop style system with menus to allow people to select various options and settings down the line. In 1.6.0, this will solely be utilized to create, join, and leave lobbies.
- A player can only be in a single lobby, if any.
- A player can only be 'hosting' a single lobby at a time. This would allow icons with player colors on them to be used to signify a lobby.
- A rework to the board system. Players will need to be able to cycle the boards. There is a need to switch to a one board per player setup, to fix a replay only display issue due to the use of `GetLocalPlayer ()`. This probably deserves its own issue.
- Things like settings, rank, etc. are tied to a lobby. Perhaps a better choice of words is needed, but it fits. A lobby has multiple player games. Each player has only one game.

**Comment by nvs** [comment]
This originally was planned for 1.6. However, with Lua on the horizon it would be easier to just rewrite the map and address this then. So, 1.7 it is.
